parsley/shared/src/main/scala/parsley/errors/combinator.scala

```diff
@@ -570,4 +584,16 @@ object combinator {
         }
         // $COVERAGE-ON$
     }
+
+    private [parsley] def filterWith[A](p: Parsley[A])(f: A => Boolean, err: Parsley[((A, Int)) => Nothing]): Parsley[A] = {
```

Comment:
Problem is, same thing applies here, but this is not in a good position to action on it. It's either in the intrinsic, or in a `select` argument position.
Perhaps we do really have to make an `ErrGen[A]` type

```suggestion
    private [parsley] def filterWith[A](p: Parsley[A])(f: A => Boolean, err: ErrGen[A]): Parsley[A] = {
```

Then have something like

```scala
abstract class ErrGen[A] { 
    def asParser: Parsley[((A, Int)) => Nothing] // use at your own risk
    def asCombinator(p: Parsley[(A, Int)]): Parsley[Nothing] // safe to use
}
```

Or something to that effect.